core/src/validator.rs

```diff
@@ -2181,6 +2183,63 @@ fn maybe_warp_slot(
     Ok(())
 }
 
+/// Returns the starting slot at which the blockstore should be scanned for
+/// shreds with an incorrect shred version, or None if the check is unnecessary
+fn should_check_blockstore_for_incorrect_shred_version(
```

Comment:
nit: since this is run only on startup it would be nice to have a log line or two here for debugging showing why we skipped or decided to perform the scan with a certain slot.

Reply:
Done in https://github.com/anza-xyz/agave/pull/2851/commits/5af033367b279235321153d224f870207affbb53